### PR TITLE
ADD: Endpoint for multi api requests

### PR DESCRIPTION
**Pull request by sihrc**

### CODE REVIEW

File: tests/test_remote.py [hunk 1]
```diff
@@ -107,6 +108,25 @@ def test_batch_language(self):
         self.assertTrue(isinstance(response, list))
         self.assertTrue(response[0]['English'] > 0.25)
 
+    def test_multi_api_image(self):
+        test_data = [generate_array((48,48))]
+        response = batch_image(test_data, apis=config.IMAGE_APIS, api_key=self.api_key)
+
+        print response
```
Comment (madisonmay) [inline tests/test_remote.py]: Strip out this print.

File: tests/test_remote.py [hunk 2]
```diff
@@ -107,6 +108,25 @@ def test_batch_language(self):
...
+        pass
+
+    def test_multi_api_text(self):
+        test_data = ['clearly an english sentence']
+        response = batch_text(test_data, apis=config.TEXT_APIS, api_key=self.api_key)
+
+        pass
```
Comment (madisonmay) [inline tests/test_remote.py]: Pass isn't needed.

File: indicoio/utils/multi.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        raise ValueError("%s are not valid %s APIs. Please reference the available APIs below:\n%s"
+                            % (", ".join(invalid_apis), type, ", ".join(available))
+                        )
+    # Convert client api names to server names before sending request
+    apis = map(CLIENT_SERVER_MAP.get, apis)
+    result = api_handler(data, apis=apis, **kwargs)
+
+    try:
+        # Parse out the results to a dicionary of api: result
+        result = dict((SERVER_CLIENT_MAP[api], res.get("results", res.get("error", False)))
```
Comment (madisonmay) [inline indicoio/utils/multi.py]: Nitpicking here, but might be worth splitting this up just a bit, as it's a fair amount of logic for a dictionary comprehension.

```
def parsed_response(response):
    return response.get('results') or response.get('error', False)

result = dict((SERVER_CLIENT_MAP[api], parsed_response(res) for api, res in result.iteritems())
``

File: indicoio/utils/multi.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
+from indicoio.config import TEXT_APIS, IMAGE_APIS, API_NAMES
+from indicoio.utils import api_handler
+
+
+CLIENT_SERVER_MAP = dict((api, api.strip().replace("_", "").lower()) for api in API_NAMES)
+CLIENT_SERVER_MAP["posneg"] = "sentiment" # Is this necessary?
```
Comment (madisonmay) [inline indicoio/utils/multi.py]: I think we can safely remove this at this point.  Just add a note in the CHANGES.txt file when we push to PyPI and we should be fine.

**Comment by madisonmay** [comment]
Looks great!  Just the tiniest of fixes, a test added for the non-batch text and image functions, and we should be good to go.

The docstrings are a really nice touch by the way.

File: indicoio/utils/multi.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
+from indicoio.config import TEXT_APIS, IMAGE_APIS, API_NAMES
+from indicoio.utils import api_handler
+
+
+CLIENT_SERVER_MAP = dict((api, api.strip().replace("_", "").lower()) for api in API_NAMES)
+SERVER_CLIENT_MAP = dict((v, k) for k, v in CLIENT_SERVER_MAP.iteritems())
+CLIENT_SERVER_MAP["posneg"] = "sentiment" # Is this necessary?
```
Comment (sihrc) [inline indicoio/utils/multi.py]: @madisonmay I'm not sure what the deal is with posneg. Is it an endpoint we want supported? (It's basically sentiment, right?)

Comment (madisonmay) [inline indicoio/utils/multi.py]: Yes, it's "legacy" code -- `sentiment` started out as `posneg`, we changed the name but left an alias for anyone who was using the original format (that was last fall).  We can safely remove it at this point.

**Comment by madisonmay** [comment]
LGTM!  Nice work.  I'm going to ensure everything is good to go on the remote end, and then this should get merged in.

File: indicoio/utils/multi.py [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+            for api, res in result.iteritems())
+    except KeyError:
+        for api in result:
+            if "error" in result[api]:
+                raise ValueError(result[api]["error"])
+        raise Exception("Sorry, %s API returned an unexpected response:\n%s" % (api, result[api]))
+
+
+def predict_text(input_text, apis, cloud=None, batch=False, api_key=None, **kwargs):
+    """
```
Comment (madisonmay) [inline indicoio/utils/multi.py]: I think it makes sense to default to the set of all text APIs.

File: indicoio/utils/multi.py [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        available = TEXT_APIS,
+        cloud=cloud,
+        batch=batch,
+        api_key=api_key,
+        apis=apis,
+        **kwargs)
+
+
+def predict_image(image, apis, cloud=None, batch=False, api_key=None, **kwargs):
+    """
```
Comment (madisonmay) [inline indicoio/utils/multi.py]: Ditto -- let's default to passing these through all of the image APIs.

File: indicoio/utils/multi.py (hunk 6 above)
Comment (madisonmay) [inline indicoio/utils/multi.py]: i.e. 

```
def predict_text(input_text, apis=TEXT_APIS, cloud=None, batch=False, api_key=None, **kwargs):
```

**Comment by madisonmay** [comment]
Re-running the build to ensure nothing has changed on the remote that would require modification of client code.

**Comment by madisonmay** [comment]
Looks like we do need to take another look -- some of the more recent changes to the server code have broken the clients.

**Comment by madisonmay** [comment]
Has to do with how we're handling batch requests.  Could you look into this tomorrow morning @sihrc ?

**Comment by madisonmay** [comment]
I think the commit above ^ should do the trick.  Are we fine with this response format?  I think it makes sense since it's most similar to the raw response we're getting from the APIs.

**Comment by madisonmay** [comment]
Tests pass, code looks good!  Merging into development.

**Pull request merged**